File: src/main/java/com/ject/studytrip/pomodoro/application/dto/PomodoroInfo.java
package com.ject.studytrip.pomodoro.application.dto;

import com.ject.studytrip.pomodoro.domain.model.Pomodoro;

public record PomodoroInfo(Long pomodoroId, int focusDurationInMinute) {
    public static PomodoroInfo from(Pomodoro pomodoro) {
        return new PomodoroInfo(pomodoro.getId(), pomodoro.getFocusDurationInSeconds() / 60);
    }
}

File: src/main/kotlin/com/ject/studytrip/pomodoro/application/dto/PomodoroInfo.kt
package com.ject.studytrip.pomodoro.application.dto

import com.ject.studytrip.global.util.EntityExtensions.requireId
import com.ject.studytrip.pomodoro.domain.model.Pomodoro

data class PomodoroInfo(
    val pomodoroId: Long,
    val focusDurationInMinute: Int,
    val focusSessionCount: Int,
) {
    companion object {
        fun from(pomodoro: Pomodoro): PomodoroInfo =
            PomodoroInfo(
                pomodoro.id.requireId(),
                pomodoro.focusDurationInSeconds / 60,
                pomodoro.focusSessionCount,
            )
    }
}
